src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs

```diff
@@ -151,6 +149,22 @@ public static OrchardCoreBuilder AddDataAccess(this OrchardCoreBuilder builder)
                 services.AddScoped<IDocumentStore, DocumentStore>();
                 services.AddSingleton<IFileDocumentStore, FileDocumentStore>();
                 services.AddTransient<IDbConnectionAccessor, DbConnectionAccessor>();
+            })
+            .InitializeServices(async serviceProvider =>
```

Comment:
The name doesn't make it easy to understand what it does and when it's called.

Reply:
@sebastienros 

I'm open to any name ;) It is executed when a shell container is built, so after all `ConfigureServices()` and of course before all `Configure()`used to build the tenant pipeline. It allows to init some services in an async way, here I use it to initialize the `IStore`, see my 1st comment https://github.com/OrchardCMS/OrchardCore/pull/13169#issue-1561147273

    .InitializeServices(async serviceProvider =>
    {
        var store = serviceProvider.GetService<IStore>();
        if (store == null)
        {
            return;
        }

        await store.InitializeAsync();

        ...
    });